adapters/IronSource/IronSourceAdapter/ISMediationManager.m

```diff
@@ -20,425 +20,578 @@
-        @"Instance already loaded. Couldn't load another one in the same time!");
-    [adapterDelegate interstitialDidFailToLoadWithError:error instanceId:instanceID];
-  }
+    id<GADMAdapterIronSourceInterstitialDelegate> adapterDelegate = delegate;
+    if (adapterDelegate == nil) {
+        [GADMAdapterIronSourceUtils
+         onLog:[NSString
+                stringWithFormat:@"requestInterstitialAdWithDelegate adapterDelegate is null."]];
+        return;
+    }
+    
+    if ([self canLoadInterstitialInstance:instanceID]) {
+        [self setInterstitialDelegate:adapterDelegate toInstanceState:GADMAdapterIronSourceInstanceStateLocked];
+        [self addInterstitialDelegate:adapterDelegate forInstanceID:instanceID];
+        [GADMAdapterIronSourceUtils
+         onLog:[NSString
+                stringWithFormat:@"ISMediationManager - loadInterstitial  for instance Id %@",
+                instanceID]];
+        [IronSource loadISDemandOnlyInterstitial:instanceID];
+    } else {
+        NSError *error = GADMAdapterIronSourceErrorWithCodeAndDescription(
+                                                                          GADMAdapterIronSourceErrorAdAlreadyLoaded,
+                                                                          @"Instance already loaded. Couldn't load another one in the same time!");
+        [adapterDelegate interstitialDidFailToLoadWithError:error instanceId:instanceID];
+    }
 }
 
 - (void)presentInterstitialAdFromViewController:(nonnull UIViewController *)viewController
                                      instanceID:(nonnull NSString *)instanceID {
-  [GADMAdapterIronSourceUtils
-      onLog:[NSString stringWithFormat:@"ISMediationManager - showInterstitial for instance Id %@",
-                                       instanceID]];
-  [IronSource showISDemandOnlyInterstitial:viewController instanceId:instanceID];
+    [GADMAdapterIronSourceUtils
+     onLog:[NSString stringWithFormat:@"ISMediationManager - showInterstitial for instance Id %@",
+            instanceID]];
+    [IronSource showISDemandOnlyInterstitial:viewController instanceId:instanceID];
+}
+
+- (void)loadBannerAdWithDelegate:(nonnull id<GADMAdapterIronSourceBannerDelegate>)delegate
+                  viewController:(nonnull UIViewController *)viewController
+                      instanceID:(nonnull NSString *)instanceID
+                      bannerSize:(nonnull ISBannerSize*)size {
+    id<GADMAdapterIronSourceBannerDelegate> adapterDelegate = delegate;
+    if (adapterDelegate == nil) {
+        [GADMAdapterIronSourceUtils
+         onLog:[NSString
+                stringWithFormat:@"requestBannerAdWithDelegate adapterDelegate is null."]];
+        return;
+    }
+
+        [self setBannerDelegate:adapterDelegate toInstanceState:GADMAdapterIronSourceInstanceStateLocked];
+        [self addBannerDelegate:adapterDelegate forInstanceID:instanceID];
+        
+        [IronSource destroyISDemandOnlyBannerWithInstanceId:instanceID];
```

Comment:
This destroys the previous banner ad. I assume IronSource does not support multiple banner ads being simultaneously loaded using the same Instance ID? Please confirm if this is the case.

If so, then the adapter should handle this and throw an error if there is already an ad loading for an IronSource instance Id.

Reply:
Its correct IronSource for now does not support multiple banner ads being simultaneously loaded.
I have changed the logic to destroy the instance once another instance is shown